packages/discord.js/src/structures/MessageMentions.js

```diff
@@ -184,17 +184,22 @@ class MessageMentions {
    * @returns {boolean}
    */
   has(data, { ignoreDirect = false, ignoreRoles = false, ignoreEveryone = false } = {}) {
-    if (!ignoreEveryone && this.everyone) return true;
-    const { GuildMember } = require('./GuildMember');
-    if (!ignoreRoles && data instanceof GuildMember) {
-      for (const role of this.roles.values()) if (data.roles.cache.has(role.id)) return true;
+    const user = this.client.users.resolve(data),
+      role = this.guild?.roles.resolve(data) ?? null,
+      channel = this.client.channels.resolve(data),
+      isDirect = user ?? role ?? channel;
```

Comment:
For consistency with the rest of the lib
```suggestion
    const user = this.client.users.resolve(data);
    const role = this.guild?.roles.resolve(data) ?? null;
    const channel = this.client.channels.resolve(data);
    const isDirect = user ?? role ?? channel;
```